sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -910,18 +911,26 @@ Mono<VerifyResult> verifyData(SignatureAlgorithm algorithm, byte[] data, byte[]
         });
     }
 
-    private Mono<Boolean> ensureValidKeyAvailable() {
+    private Mono<Boolean> isValidKeyLocallyAvailable() {
         boolean keyNotAvailable = (key == null && keyCollection != null);
-        boolean keyNotValid = (key != null && !key.isValid());
 
-        if (keyNotAvailable || keyNotValid) {
+        if (keyNotAvailable) {
             if (keyCollection.equals(CryptographyClientImpl.SECRETS_COLLECTION)) {
                 return getSecretKey().map(jsonWebKey -> {
-                    key = (jsonWebKey);
+                    key = jsonWebKey;
 
                     if (key.isValid()) {
                         if (localKeyCryptographyClient == null) {
-                            localKeyCryptographyClient = initializeCryptoClient(key, implClient);
+                            try {
+                                localKeyCryptographyClient = initializeCryptoClient(key, implClient);
+                            } catch (RuntimeException e) {
+                                localOperationNotSupported = true;
+
+                                LOGGER.logExceptionAsWarning(
```

Comment:
Since we aren't throwing or returning this exception there is no need to wrap this in a `RuntimeException` and we should call `LOGGER.warning("Defaulting to service use for cryptographic operations.", e)`